php/libraries/User.class.inc

```diff
@@ -341,6 +341,34 @@ class User extends UserPermissions implements
         return explode(';', $projects);
     }
 
+    /**
+     * Get the visits the User has access to
+     * through their projects and their cohorts.
+     *
+     * @return array of study visits in the format array('VL' => 'VL')
+     *         where VL is the visit label
+     */
+    public function getVisits(): array
```

Comment:
@skarya22 this is purely out of personal preference and not at all blocking:

I would use this opportunity to create a `getVisits()` function in the `Project` class which knows how to get the visits itself instead of doing these loops through projects...cohorts...visits.

That being said, the efficiency of the code below is basically one query per project per cohort, if you choose to add the function to the project class, you could choose to use `getVisitsProjectCohort()` from the `VisitController` class to make it an efficiency of one query per project. ORR if you choose not to create the Projec::getvisits() function you could use `getVisitsProjectCohort()` from the `VisitController` class here directly to make it an efficiency of just 1 query!

I'll leave it up to you